Update Elasticsearch client and add support for 6.0.0-beta2

Update to latest stable Elasticsearch client (currently 5.5.2), add support for latest Elasticsearch 6.0.0 pre-release (currently 6.0.0-beta2) and restore automated tests for Elasticsearch 1.x (requires Docker). Also cleanup tests as described below.

* Use `indes.search.hostname` argument instead of `is.external.es` for testing against an externally-manged server instance
* Remove lingering transport client references from ElasticSearchSetup
* Consolidate Elasticsearch test configuration to remove redundancy
* Cleanup ElasticSearchConfigTest
* Fix typo in ElasticSearchMultiTypeIndexTest name

I just pulled this in and tried to build.. Got this error:

```
[janusgraph]$ mvn clean install -Pjanusgraph-release -Dgpg.skip=true -DskipTests=true 
[INFO] Scanning for projects...
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[ERROR] 'dependencies.dependency.version' for org.elasticsearch.distribution.zip:elasticsearch:zip must be a valid version but is '${elasticsearch.test.version}'. @ org.janusgraph:janusgraph-es:[unknown-version], /home/rdale/git/forks/janusgraph/janusgraph/janusgraph-es/pom.xml, line 68, column 22
 @ 
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project org.janusgraph:janusgraph-es:0.2.0-SNAPSHOT (/home/rdale/git/forks/janusgraph/janusgraph/janusgraph-es/pom.xml) has 1 error
[ERROR]     'dependencies.dependency.version' for org.elasticsearch.distribution.zip:elasticsearch:zip must be a valid version but is '${elasticsearch.test.version}'. @ org.janusgraph:janusgraph-es:[unknown-version], /home/rdale/git/forks/janusgraph/janusgraph/janusgraph-es/pom.xml, line 68, column 22
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
```

--- a/janusgraph-es/pom.xml
+++ b/janusgraph-es/pom.xml
@@ -15,6 +15,9 @@
         <elasticsearch2.test.version>2.4.6</elasticsearch2.test.version>
         <elasticsearch5.test.version>5.6.0</elasticsearch5.test.version>
         <elasticsearch6.test.version>6.0.0-beta2</elasticsearch6.test.version>
+        <elasticsearch.test.version>${elasticsearch5.test.version}</elasticsearch.test.version>
+        <!-- Enabling Groovy dynamic scripting is not required from ES 5.x -->
+        <elasticsearch.test.groovy.inline/>
         <skip.es.test>${skipTests}</skip.es.test>
         <es.docker.image>docker.elastic.co/elasticsearch/elasticsearch</es.docker.image>
         <elasticsearch.docker.test.version>${elasticsearch.test.version}</elasticsearch.docker.test.version>

